*** Makefile.in.orig	Fri Jan 18 16:23:44 2002
--- Makefile.in	Fri Jan 18 16:25:00 2002
***************
*** 49,56 ****
  # These definitions have to come before the rules that expand them.
  # The tex{ini,0,1,2}.c files are created by splitup, run as part of convert.
  ptex_c = texini.c tex0.c tex1.c tex2.c
! @PLAIN@ptex_o = texini.o tex0.o tex1.o tex2.o ptexextra-plain.o
! @SRCSP@ptex_o = texini.o tex0.o tex1.o tex2.o ptexextra-src-special.o
  plib_o = kanji.o version.o usage.o
  
  # Prevent Make from deleting the intermediate forms.
--- 49,56 ----
  # These definitions have to come before the rules that expand them.
  # The tex{ini,0,1,2}.c files are created by splitup, run as part of convert.
  ptex_c = texini.c tex0.c tex1.c tex2.c
! #@PLAIN@ptex_o = texini.o tex0.o tex1.o tex2.o ptexextra-plain.o
! ptex_o = texini.o tex0.o tex1.o tex2.o ptexextra-src-special.o
  plib_o = kanji.o version.o usage.o
  
  # Prevent Make from deleting the intermediate forms.
***************
*** 79,86 ****
  tex0.o: tex0.c texcoerce.h texd.h kanji.h
  tex1.o: tex1.c texcoerce.h texd.h kanji.h
  tex2.o: tex2.c texcoerce.h texd.h kanji.h
! @PLAIN@ptexextra-plain.o: ptexextra-plain.c texcoerce.h texd.h kanji.h texmfmp-help.h
! @SRCSP@ptexextra-src-special.o: ptexextra-src-special.c texcoerce.h texd.h kanji.h texmfmp-help.h
  
  $(ptex_c) texcoerce.h texd.h: ptex.p $(web2c_aux) $(web2c_programs)
  	$(web2c) ptex
--- 79,86 ----
  tex0.o: tex0.c texcoerce.h texd.h kanji.h
  tex1.o: tex1.c texcoerce.h texd.h kanji.h
  tex2.o: tex2.c texcoerce.h texd.h kanji.h
! #@PLAIN@ptexextra-plain.o: ptexextra-plain.c texcoerce.h texd.h kanji.h texmfmp-help.h
! ptexextra-src-special.o: ptexextra-src-special.c texcoerce.h texd.h kanji.h texmfmp-help.h
  
  $(ptex_c) texcoerce.h texd.h: ptex.p $(web2c_aux) $(web2c_programs)
  	$(web2c) ptex
***************
*** 90,99 ****
  ptex.web: $(srcdir)/tex.web $(srcdir)/tex.ch
  	$(srcdir)/tie -m ptex.web $(srcdir)/tex.web $(srcdir)/tex.ch
  
- @PLAIN@srcsp=
- @SRCSP@srcsp=$(srcdir)/tex-src-special.ch $(srcdir)/tex-c-style.ch
  ptex.ch: $(srcdir)/tie ptex.web ptex-base.ch
! 	$(srcdir)/tie -c ptex.ch ptex.web ptex-base.ch $(srcsp)
  
  ### TFtoPL
  tftopl: tftopl.o $(plib_o) $(kpathsea) $(proglib)
--- 90,97 ----
  ptex.web: $(srcdir)/tex.web $(srcdir)/tex.ch
  	$(srcdir)/tie -m ptex.web $(srcdir)/tex.web $(srcdir)/tex.ch
  
  ptex.ch: $(srcdir)/tie ptex.web ptex-base.ch
! 	$(srcdir)/tie -c ptex.ch ptex.web ptex-base.ch
  
  ### TFtoPL
  tftopl: tftopl.o $(plib_o) $(kpathsea) $(proglib)
*** tftopl.ch.orig	Fri Jan 18 16:17:57 2002
--- tftopl.ch	Fri Jan 18 16:11:00 2002
***************
*** 337,347 ****
    end;
  @z
  
! @x
!       usage (0, TFTOPL_HELP);
! @y
!       usage_help (PTEX_TFTOPL_HELP);
! @z
  
  @x [99] l.1751 - pTeX:
  @* Index.
--- 337,347 ----
    end;
  @z
  
! %@x
! %      usage (0, TFTOPL_HELP);
! %@y
! %      usage_help (PTEX_TFTOPL_HELP);
! %@z
  
  @x [99] l.1751 - pTeX:
  @* Index.
*** pltotf.ch.orig	Fri Jan 18 16:18:04 2002
--- pltotf.ch	Fri Jan 18 16:11:07 2002
***************
*** 390,400 ****
  end;
  @z
  
! @x
!       usage (0, PLTOTF_HELP);
! @y
!       usage_help (PTEX_PLTOTF_HELP);
! @z
  
  @x [148] l.2620 - pTeX:
  @* Index.
--- 390,400 ----
  end;
  @z
  
! %@x
! %      usage (0, PLTOTF_HELP);
! %@y
! %      usage_help (PTEX_PLTOTF_HELP);
! %@z
  
  @x [148] l.2620 - pTeX:
  @* Index.
*** pdvitype.ch.orig	Fri Jan 18 16:18:17 2002
--- pdvitype.ch	Fri Jan 18 16:11:53 2002
***************
*** 1003,1009 ****
        {End of arguments; we exit the loop below.} ;
  
      end else if getopt_return_val = "?" then begin
!       usage (1, 'pdvitype');
  
      end else if argument_is ('help') then begin
        usage_help (PDVITYPE_HELP);
--- 1003,1009 ----
        {End of arguments; we exit the loop below.} ;
  
      end else if getopt_return_val = "?" then begin
!       usage ('pdvitype');
  
      end else if argument_is ('help') then begin
        usage_help (PDVITYPE_HELP);
***************
*** 1036,1042 ****
    {Now |optind| is the index of first non-option on the command line.}
    if (optind + 1 <> argc) then begin
      write_ln (stderr, 'pdvitype: Need exactly one file argument.');
!     usage (1, 'pdvitype');
    end;
  end;
  
--- 1036,1042 ----
    {Now |optind| is the index of first non-option on the command line.}
    if (optind + 1 <> argc) then begin
      write_ln (stderr, 'pdvitype: Need exactly one file argument.');
!     usage ('pdvitype');
    end;
  end;
  
*** jbibtex.ch.orig	Fri Jan 18 16:18:24 2002
--- jbibtex.ch	Fri Jan 18 16:12:01 2002
***************
*** 1485,1491 ****
        {End of arguments; we exit the loop below.} ;
  
      end else if getopt_return_val = "?" then begin
!       usage (1, 'jbibtex');
  
      end else if argument_is ('min-crossrefs') then begin
        min_crossrefs := atoi (optarg);
--- 1485,1491 ----
        {End of arguments; we exit the loop below.} ;
  
      end else if getopt_return_val = "?" then begin
!       usage ('jbibtex');
  
      end else if argument_is ('min-crossrefs') then begin
        min_crossrefs := atoi (optarg);
***************
*** 1503,1509 ****
     We must have one remaining argument.}
    if (optind + 1 <> argc) then begin
      write_ln (stderr, 'jbibtex: Need exactly one file argument.');
!     usage (1, 'jbibtex');
    end;
  end;
  
--- 1503,1509 ----
     We must have one remaining argument.}
    if (optind + 1 <> argc) then begin
      write_ln (stderr, 'jbibtex: Need exactly one file argument.');
!     usage ('jbibtex');
    end;
  end;
  
*** jbibtex.defines.orig	Fri Jan 18 16:18:29 2002
--- jbibtex.defines	Fri Jan 18 16:12:25 2002
***************
*** 1,4 ****
  @define function inputline();
  @define procedure initkanji;
  @define const JBIBTEXHELP;
- @define procedure usagehelp();
--- 1,3 ----
*** kanji.defines.orig	Fri Jan 18 16:18:38 2002
--- kanji.defines	Fri Jan 18 16:05:48 2002
***************
*** 21,25 ****
  @define const PTEXTFTOPLHELP;
  @define const PTEXPLTOTFHELP;
  @define const PDVITYPEHELP;
- 
- @define procedure usagehelp();
--- 21,23 ----
*** usage.c.orig	Fri Jan 18 16:20:04 2002
--- usage.c	Fri Jan 18 16:20:35 2002
***************
*** 3,12 ****
  
  /*** lib/usage.c:  Output a help message (from help.h). ***/
  void
! usage P2C(int, status,  const_string, str)
  {
    fprintf (stderr, "Try `%s --help' for more information.\n", str);
!   uexit (status);
  }
  
  /* Call usage if the program exits by printing the help message.
--- 3,12 ----
  
  /*** lib/usage.c:  Output a help message (from help.h). ***/
  void
! usage P1C(const_string, str)
  {
    fprintf (stderr, "Try `%s --help' for more information.\n", str);
!   uexit (1);
  }
  
  /* Call usage if the program exits by printing the help message.
*** ptex-base.ch.orig	Fri Jan 18 16:26:10 2002
--- ptex-base.ch	Fri Jan 18 16:01:55 2002
***************
*** 3941,3949 ****
--- 3941,3955 ----
  
  @x [46.1034] l.20072 - pTeX: disp_node
  @<Append character |cur_chr|...@>=
+ if ((head=tail) and (mode>0)) then begin
+   if (insert_src_special_auto) then append_src_special;
+ end;
  adjust_space_factor;@/
  @y
  @<Append character |cur_chr|...@>=
+ if ((head=tail) and (mode>0)) then begin
+   if (insert_src_special_auto) then append_src_special;
+ end;
  adjust_space_factor;@/
  if direction=dir_tate then disp:=t_baseline_shift else disp:=y_baseline_shift;
  @<Append |disp_node| at begin of displace area@>;
***************
*** 5280,5285 ****
--- 5286,5297 ----
  show_mode: @<Show the current japanese processing mode@>;
  @z
  
+ @x
+ font_info:=xmalloc_array(fmemory_word, font_mem_size);
+ @y
+ font_info:=xmalloc_array(memory_word, font_mem_size);
+ @z
+ 
  @x [50.1320] l.24357 - pTeX:
  @ @<Dump the array info for internal font number |k|@>=
  begin
***************
*** 5305,5319 ****
  @y
  @<Undump the array info for internal font number |k|@>=
  begin {Allocate the font arrays}
! xmalloc_array(font_dir, font_max);
! xmalloc_array(font_num_ext, font_max);
  @z
  
  @x [50.1322] l.24409 - pTeX:
! xmalloc_array(char_base, font_max);
  @y
! xmalloc_array(ctype_base, font_max);
! xmalloc_array(char_base, font_max);
  @z
  
  @x [50.1322] l.24419 - pTeX:
--- 5317,5331 ----
  @y
  @<Undump the array info for internal font number |k|@>=
  begin {Allocate the font arrays}
! font_dir:=xmalloc_array(eight_bits, font_max);
! font_num_ext:=xmalloc_array(integer, font_max);
  @z
  
  @x [50.1322] l.24409 - pTeX:
! char_base:=xmalloc_array(integer, font_max);
  @y
! ctype_base:=xmalloc_array(integer, font_max);
! char_base:=xmalloc_array(integer, font_max);
  @z
  
  @x [50.1322] l.24419 - pTeX:
***************
*** 5331,5349 ****
  undump_things(char_base[null_font], font_ptr+1-null_font);
  @z
  
  @x [51.1337] l.24977 - pTeX:
!   xmalloc_array(font_check, font_max);
  @y
!   xmalloc_array(font_dir, font_max);
!   xmalloc_array(font_num_ext, font_max);
!   xmalloc_array(font_check, font_max);
  @z
  
  @x [51.1337] l.24991 - pTeX:
!   xmalloc_array(char_base, font_max);
  @y
!   xmalloc_array(ctype_base, font_max);
!   xmalloc_array(char_base, font_max);
  @z
  
  @x [51.1337] l.25001 - pTeX:
--- 5343,5367 ----
  undump_things(char_base[null_font], font_ptr+1-null_font);
  @z
  
+ @x
+   font_info:=xmalloc_array (fmemory_word, font_mem_size);
+ @y
+   font_info:=xmalloc_array (memory_word, font_mem_size);
+ @z
+ 
  @x [51.1337] l.24977 - pTeX:
!   font_check:=xmalloc_array(four_quarters, font_max);
  @y
!   font_dir:=xmalloc_array(eight_bits, font_max);
!   font_num_ext:=xmalloc_array(integer, font_max);
!   font_check:=xmalloc_array(four_quarters, font_max);
  @z
  
  @x [51.1337] l.24991 - pTeX:
!   char_base:=xmalloc_array(integer, font_max);
  @y
!   ctype_base:=xmalloc_array(integer, font_max);
!   char_base:=xmalloc_array(integer, font_max);
  @z
  
  @x [51.1337] l.25001 - pTeX:
*** ptexextra-src-special.c.orig	Fri Jan 18 17:42:41 2002
--- ptexextra-src-special.c	Fri Jan 18 17:42:51 2002
***************
*** 737,743 ****
        break;
  
      if (g == '?') { /* Unknown option.  */
!       usage (1, argv[0]);
      }
  
      assert (g == 0); /* We have no short option names.  */
--- 737,743 ----
        break;
  
      if (g == '?') { /* Unknown option.  */
!       usage (argv[0]);
      }
  
      assert (g == 0); /* We have no short option names.  */
*** configure.orig	Fri Jan 18 21:20:55 2002
--- configure	Fri Jan 18 21:21:12 2002
***************
*** 47,58 ****
  sed -f $CONFIGSED kanji.h.in > kanji.h
  
  echo "texmfmp-help.h"
! if [ x$srcsp = xyes ]; then
! 	cat texmfmp-help.h.in > texmfmp-help.h
! else
! 	sed -e '/src-special/d' texmfmp-help.h.in > texmfmp-help.h
! fi
! 
  
  # Makefile の作成
  CONFIG_FILES=Makefile
--- 47,53 ----
  sed -f $CONFIGSED kanji.h.in > kanji.h
  
  echo "texmfmp-help.h"
! cat texmfmp-help.h.in > texmfmp-help.h
  
  # Makefile の作成
  CONFIG_FILES=Makefile
